clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -990,16 +990,18 @@ UnsentRequest buildRequest() {
             }
 
             ShareAcknowledgeRequest.Builder requestBuilder = sessionHandler.newShareAcknowledgeBuilder(groupId, fetchConfig);
```

Comment:
It seems to me that `ShareSessionHandler.addPartitionToFetch` saves partition and acknowledgement state into the handler to be built by the request builder. In the event that the request builder cannot be built, such as because a new share session needs to be created, the partition and acknowledgement state is still left in the share session. I worry that it is possible this could lead to an erroneous request in some situations when the session handler is used again when a new connection has been created. I think that it would be prudent to clear out `nextAcknowledgements` and `nextPartitions` in `ShareSessionHandler.newShareAcknowledgeBuilder` in the case where it returns null.

Reply:
Yeah, right. The handler does not clear the acknowledgements which can be a problem when future requests are built. I have added code to clear the `nextPartitions` and `nextAcknowledgements` now and added a test for the same.
Thanks for catching this.